### PR TITLE
Fix list view icons with custom links

### PR DESCRIPTION
**Pull request by skurvish**
Created a new layout combining the custom links with the listicon-tips for the case of a custom link attached to an icon. See http://fabrikar.com/forums/index.php?threads/icon-link-in-list-to-custom-php-script.48785/ for details

### CODE REVIEW

**Comment by Sophist-UK** [comment]
At first sight the code looks good. Can you post before and after screen shots and HTML as before?

**Comment by skurvish** [comment]
The screen shots are identical in either case, what is critical is the underlying html. Here is the html before the change: 
image and here is the html after the change: 
image

**Comment by Sophist-UK** [comment]
That looks cool. Not sure when Hugh will be able to merge it into the main Fabrik code, but I would be happy to merge it into my alternative "Update from Github" repo if you are willing to support it there.

See https://github.com/Sophist-UK/joomla_fabrik/wiki

**Comment by cheesegrits** [comment]
It's merged in my local test branch, I'll merge it to master when I've finished testing.

File: components/com_fabrik/models/list.php [hunk 1]
```diff
@@ -2320,12 +2320,23 @@ public function _addLink($data, &$elementModel, $row, $repeatCounter = 0)
...
+		$isIcon = FALSE;
+		
+		/* Check if the data component is html, if it is an icon lnk and image parse it out */
+		if (class_exists('DOMDocument')) {
+			$html = new DOMDocument;
+			$html->loadHTML($data);
+			$a = $html->getElementsByTagName('a')->item(0);
+			$img = $html->getElementsByTagName('img')->item(0);
+			if (isset($html) && isset($a) && isset($img))
+				$isIcon = TRUE;
```
Comment (cheesegrits) [inline components/com_fabrik/models/list.php]: Put {} around the body.  I know it's not technically required for a single statement.

File: components/com_fabrik/models/list.php [hunk 2]
```diff
@@ -2320,12 +2320,23 @@ public function _addLink($data, &$elementModel, $row, $repeatCounter = 0)
...
+		}
 
 		$paths = array(
 			COM_FABRIK_FRONTEND . '/views/list/tmpl/' . $this->getTmpl() . '/layouts/element/' . $elementModel->getFullName(true, false)
 		);
 
-		$layout                  = $this->getLayout('element.fabrik-element-custom-link', $paths);
+		$layout = $isIcon ? $this->getLayout('element.fabrik-element-custom-icon-link', $paths) : $this->getLayout('element.fabrik-element-custom-link', $paths);
```
Comment (cheesegrits) [inline components/com_fabrik/models/list.php]: Avoid long lines, or PHP Storm will perpetually bitch at me about it.  So only use inline condition for short stuff.  Put this in a normal if/else.

File: components/com_fabrik/models/list.php [hunk 3]
```diff
@@ -2320,12 +2320,23 @@ public function _addLink($data, &$elementModel, $row, $repeatCounter = 0)
 		$rowId = $this->getSlug($row);
 		$isAjax = $this->isAjaxLinks() ? '1' : '0';
 		$isCustom = $customLink !== '';
+		$isIcon = FALSE;
```
Comment (cheesegrits) [inline components/com_fabrik/models/list.php]: lower case

File: components/com_fabrik/models/list.php (hunk 1 above)
Comment (cheesegrits) [inline components/com_fabrik/models/list.php]: lower case

**Pull request merged**